mmeval/metrics/perplexity.py

```diff
@@ -0,0 +1,281 @@
+    language model.It is defined as 2 to the power of the cross-entropy
+    loss of the model(or the negative log-likelihood of the sample).
+
+    Args:
+        ignore_labels (Union[int, List[int], None]): Integer specifying a
+            target class to ignore. If given, this class index does not
+            contribute to the returned score. Defaults to None.
+        **kwargs: Keyword parameters passed to :class:`BaseMetric`.
+
+    Examples:
+        >>> from mmeval import Perplexity
+        >>> predictions = np.asarray([1,1,1,1])
+        >>> targets = np.asarray([1,1,1,1])
+        >>> metric = Perplexity()
+        >>> result = metric(predictions, targets)
+        {'perplexity': 0.5226045319355426}
+    """
+
+    def __init__(self,
+                 ignore_labels: Union[int, List[int], None] = None,
+                 **kwargs) -> None:
+        super().__init__(**kwargs)
+        self.ignore_labels = None
+        if isinstance(ignore_labels, int):
+            ignore_labels = [ignore_labels]
+        if ignore_labels is not None:
+            for ignore_label in ignore_labels:
+                if not isinstance(ignore_label, int):
+                    raise ValueError(
+                        f'Argument `ignore_labels` expected to be `None`, `int`,`List[int]` but got {ignore_labels}'  # noqa: E501
+                    )
+            self.ignore_labels = list(set(ignore_labels))
+
+    def add(self, predictions: Sequence, targets: Sequence) -> None:  # type: ignore # yapf: disable # noqa: E501
+        """Add the intermediate results to ``self._results``.
+
+        Args:
+            predictions (Sequence): Probabilities assigned to each token in
+                a sequence with shape [batch_size, seq_len, vocab_size].
+            targets (Sequence): Ground truth values with a
+                shape [batch_size, seq_len].
+        """
+        pred_check = np.array(predictions)
+        target_check = np.array(targets)
+        if pred_check.ndim != 3:
+            raise ValueError(
+                'Input tensor `pred` is expected to have 3 dimensions, [batch_size, seq_len, vocab_size],'  # noqa: E501
+                f' but got {pred_check.ndim}.')
+        if target_check.ndim != 2:
+            raise ValueError(
+                'Input tensor `target` is expected to have 2 dimensions, [batch_size, seq_len],'  # noqa: E501
+                f' but got {target_check.ndim}.')
+        if pred_check.shape[:2] != target_check.shape:
+            raise ValueError(
+                'Input tensors `pred` and `target` are expected to have equaling first two dimensions,'  # noqa: E501
+                f' [batch_size, seq_len], but got {pred_check.shape[:2]} and {target_check.shape}.'  # noqa: E501
+            )
+        for prediction, target in zip(predictions, targets):
+            result = self._compute_result(prediction, target)
+            self._results.append(result)
```

Comment:
A suggestion that improve code readability~
```suggestion
            total_probs, count = self._compute_result(prediction, target)
            self._results.append((total_probs, count))
```